crates/analytics/src/search.rs

```diff
@@ -20,9 +20,43 @@ pub async fn msearch_results(
         OpenSearchQueryBuilder::new(OpenSearchQuery::Msearch(indexes.clone()), req.query);
 
     query_builder
-        .add_filter_clause("merchant_id".to_string(), merchant_id.to_string())
+        .add_filter_clause(
+            "merchant_id.keyword".to_string(),
+            vec![merchant_id.to_string()],
+        )
         .switch()?;
 
+    if let Some(filters) = req.filters {
+        if let Some(currency) = filters.currency {
+            if !currency.is_empty() {
+                query_builder
+                    .add_filter_clause("currency.keyword".to_string(), currency.clone())
+                    .switch()?;
+            }
+        };
+        if let Some(status) = filters.status {
+            if !status.is_empty() {
+                query_builder
+                    .add_filter_clause("status.keyword".to_string(), status.clone())
```

Comment:
is the `.keyword` part required?

Reply:
Yes, some of the fields require `.keyword`, and others don't, and we are unable to differentiate which fields require it. So adding `.keyword` for all the fields as of now.